### ISSUE TITLE
aarch64 no longer compiles, openssl differences?

### ISSUE THREAD

**Issue by poetaster** [sailfish-spotify/hutspot#116]
```
../hutspot/src/connect/spconnect.cpp: In constructor ‘SPConnect::SPConnect(QObject*)’:
../hutspot/src/connect/spconnect.cpp:41:17: error: invalid use of incomplete type ‘DH’ {aka ‘struct dh_st’}
     BN_rshift(dh->priv_key, dh->priv_key, 8);
                 ^~
In file included from /usr/include/openssl/crypto.h:25,
                 from /usr/include/openssl/bio.h:20,
                 from /usr/include/openssl/dh.h:17,
                 from ../hutspot/src/connect/spconnect.h:7,
                 from ../hutspot/src/connect/spconnect.cpp:1:
/usr/include/openssl/ossl_typ.h:106:16: note: forward declaration of ‘DH’ {aka ‘struct dh_st’}
 typedef struct dh_st DH;
                ^~~~~
../hutspot/src/connect/spconnect.cpp:41:31: error: invalid use of incomplete type ‘DH’ {aka ‘struct dh_st’}
     BN_rshift(dh->priv_key, dh->priv_key, 8);
                               ^~
In file included from /usr/include/openssl/crypto.h:25,
                 from /usr/include/openssl/bio.h:20,
                 from /usr/include/openssl/dh.h:17,
                 from ../hutspot/src/connect/spconnect.h:7,
                 from ../hutspot/src/connect/spconnect.cpp:1:
/usr/include/openssl/ossl_typ.h:106:16: note: forward declaration of ‘DH’ {aka ‘struct dh_st’}
 typedef struct dh_st DH;
                ^~~~~
../hutspot/src/connect/spconnect.cpp:44:12: error: invalid use of incomplete type ‘DH’ {aka ‘struct dh_st’}
     powm(dh->g, dh->priv_key, dh->p, dh->pub_key);
            ^~
In file included from /usr/include/openssl/crypto.h:25,
                 from /usr/include/openssl/bio.h:20,
                 from /usr/include/openssl/dh.h:17,
                 from ../hutspot/src/connect/spconnect.h:7,
                 from ../hutspot/src/connect/spconnect.cpp:1:
/usr/include/openssl/ossl_typ.h:106:16: note: forward declaration of ‘DH’ {aka ‘struct dh_st’}
 typedef struct dh_st DH;

```

**Comment by Scarrough** [comment]
I have been puzzling my way around the openssl changes from 1.0 to 1.1 and I have it compiling for arm7 but it fails at linking (like it doesn't find the openssl lib for some reason, barfs on BN_numbits, could be target specific). It is one file that changes, I can attach it here when I get home, could you let me know how you get on with it?

**Comment by vitalii-koreniev** [comment]
Unfortunately, I am too busy right now defencing from russion murderers.

**Comment by Scarrough** [comment]
That's a pretty solid reason. Hope you succeed.

**Comment by Scarrough** [comment]
I have this fixed and running on my phone. I am new to this so I am unsure how best to share this?

**Comment by poetaster** [comment]
> I have this fixed and running on my phone. I am new to this so I am unsure how best to share this?

If you forked the repo on github, you would try a Pull Request. Did you just 'clone' or also do a fork?

**Comment by Scarrough** [comment]
Just cloned I'm afraid.

**Comment by poetaster** [comment]
> Just cloned I'm afraid.

Well, you could fork it now and just copy over your changes. Commit 'em, push em and do a PR. 

Of, if it's not too many changes, on too many files, just post them here. I could help.

**Comment by Scarrough** [comment]
2 files
[hutspot-changes.zip](https://github.com/sailfish-spotify/hutspot/files/8247002/hutspot-changes.zip)
.

**Comment by poetaster** [comment]
Ok, I'll push them into my fork ....  and see if I can give you credit somehow...

**Comment by poetaster** [comment]
> 2 files [hutspot-changes.zip](https://github.com/sailfish-spotify/hutspot/files/8247002/hutspot-changes.zip) .

And a stupid question, did you test this with librespot as player? And, if so, on which architectures?

**Comment by Scarrough** [comment]
Yes, I built an rpm for the arm7 target, put it on my phone and checked it worked with Librespot on there. It behaved the same as it had before. I didn't have access to other archs with librespot installed, I checked it on the emulator (found a couple of stupid bugs like failing to use BN_new() on the public key declaration giving me a seg fault) and cleaned them up, but did not have a browser (for auth) or librespot on that VM.
